src/compiler/checker.ts

```diff
@@ -19819,7 +19819,10 @@ namespace ts {
                                 }
                                 else if (middleLength === 1 && elementFlags[startLength] & ElementFlags.Variadic) {
                                     // Middle of target is exactly one variadic element. Infer the slice between the fixed parts in the source.
-                                    inferFromTypes(isTupleType(source) ? sliceTupleType(source, startLength, sourceEndLength) : createArrayType(sourceRestType!), elementTypes[startLength]);
+                                    // If target ends in optional element(s), make a lower priority a speculative inference.

```

Comment:
If the documentation is going to be sparse on `SpeculativeTuple`, I think showing a concrete example here is worthwhile. Your explanation on the PR would be great.